### PR TITLE
[Proposal] Cleanup requirements packages

### PR DESCRIPTION
**Pull request by nissy-dev**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

I try to update the requirements packages.  How about...?

**removed packages**
- joblib : scikit-learn requires joblib, so we don't need to install manually
- requests : deepchem doesn't depend on requests
- zlib : python has zlib modules
- xgboost : this is s soft dependent package, many users don't use...

**added packages**
I added three soft dependent packages, `mdtraj`, `openmm` and `pdbfixer`. 
This is because deepchem users like bioscientist tend to use them and they are more dependent than other soft dependent packages.

### CODE REVIEW

File: recipe/run_test.py [hunk 1]
```diff
@@ -26,14 +26,6 @@ def test_rdkit_import(self):
     import rdkit
     print(rdkit.__version__)
 
-  def test_numpy_import(self):
```
Comment (nissy-dev) [inline recipe/run_test.py:29]: These test are not needed because there are hard dependent packages and deepchem importing already tested.

**Comment by simonbray** [comment]
Hi @nd-02110114, thanks for the PR. Some comments:
* openmm and pdbfixer cannot be deps, they are not available through CF.
* mdtraj is provided by CF but I think it's a bit strange to make it a dep when openmm and pdbfixer are not.
* Your other changes seem fine, I assume deepchem basic functionality will not be affected by these removals. Removing xgboost would be especially nice as it's the only dep without a windows build, so with this PR windows users would have access to a (admittedly really stripped down) conda-installable deepchem. :+1:

Thanks!

**Comment by simonbray** [comment]
I see that a section was added recently to the deepchem README dividing dependencies into hard and soft. To me it would make sense for this recipe to include only the hard ones, which would entail readding tensorflow (removed in this PR: https://github.com/conda-forge/deepchem-feedstock/pull/2) and removing rdkit as well.

**Comment by rbharath** [comment]
@simonbray's suggestions make sense to me! That is, making this recipe include the hard dependencies and not the soft dependencies. I really like the idea of getting a windows-installable conda version, so removing xgboost makes sense in particular

**Comment by nissy-dev** [comment]
Thanks you for your comments!  
I modified this recipe so that it included only the hard dependencies.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,17 +24,16 @@ requirements:
...
     - joblib
-    - scikit-learn <=0.22.1
-    - networkx
-    - xgboost
+    - numpy
     - pandas
-    - zlib
-    - requests
-    - rdkit
+    - pillow # pillow will remove from v2.4.0
```
Comment (nissy-dev) [inline recipe/meta.yaml:29]: Pillow was hard dependecies in deepchem 2.3.0, so I added it.
I think this is work around and we will remove pillow in the next 2.4.0

**Comment by simonbray** [comment]
Looks good as far as I'm concerned, but a couple of questions:
1. do scipy and numpy need to be in the requirements?
2. what about tensorflow, should this be included?

**Comment by nissy-dev** [comment]
> do scipy and numpy need to be in the requirements?

scipy and numpy are hard dependencies, so I added.  (they are not listed up in README of deepchem,  but I will update in this PR https://github.com/deepchem/deepchem/pull/1889)  
`scipy`, `numpy` and `joblib` are not necessary to be added manually because `scikit-learn` requires them.  But, I think it is safer to list up them in  the requirements considering about the possibility of breaking changes in scikit-learn

> what about tensorflow, should this be included?

This is because we led many users to install `tensorflow==1.14`. (not tensorflow==1.14.0) I think it is better to set the same version.

**Comment by simonbray** [comment]
@conda-forge-admin, please rerender

**Review by simonbray — APPROVED** [review]

**Review by rbharath — APPROVED** [review]
Changes looks good on my end!

**Pull request merged**